cmd/clusterctl/client/cluster/ownergraph.go

```diff
@@ -71,6 +78,16 @@ func GetOwnerGraph(namespace, kubeconfigPath string) (OwnerGraph, error) {
 	}
 	owners := OwnerGraph{}
 	for _, v := range graph.uidToNode {
+		// If the object is a Secret but not part of the Cluster ignore it.
+		if v.identity.Kind == "Secret" {
```

Comment:
Q: Why is the SA token secret part of the ownerGrap at all?

I thought the ownerGrap takes the root elements and then goes down with ownerRefs? That assumption is probably wrong, but it sounds off to me that the SA token is part of the grap.

Reply:
Could it be that all Secrets are added (including the ones in the provider namespace) to the graph?

(even if they are not linked to the cluster)

Reply:
The Discovery code just gets all the secrets for whatever appropriate namespace (which might include the provider namespaces, but doesn't in this test), so these default secrets will always be discovered. ownerGraph uses discovery directly which doesn't use ownerReferences for discovery, just for establishing the relationships.

Reply:
Thx!

Should we add something like this as a comment or do you think it's not necessary?

Reply:
I've added a caveat on the OwnerGraph function for now, but in future this might need to be changed to make it easier for providers to use this function.